runtime/runtime-params-estimator/src/replay/fold_db_ops.rs

```diff
@@ -0,0 +1,284 @@
+use super::cache_stats::CacheStats;
+use super::Visitor;
+use std::collections::{BTreeMap, HashMap};
+use std::io::Write;
+
+const EMPTY_STATE_ERR: &str = "states must never be empty";
+
+/// A visitor that keeps track of DB operations and aggregates it by specific labels.
+pub(super) struct FoldDbOps {
+    /// Labels at which to break aggregation, and the fields to print on each.
+    ///
+    /// If field is not available, it will be silently ignored.
+    fold_anchors: HashMap<String, Vec<String>>,
+    /// Print the DB operations that are at indentation 0, not associated with a span.
+    ///
+    /// This is a summary of the total. Instead it shows what operations have
+    /// not been covered by already reported output. To get a summary of the
+    /// total, run without folding.
+    print_top_level: bool,
+    /// Only show data for a specific smart contract, specified by account id.
+    /// Currently only applies within receipts.
+    account_filter: Option<String>,
+    /// Skip all input lines that are less indented.
+    ///
+    /// Used to apply filters and skip some parts of the trace.
+    /// Default is 0, which collects everything.
+    min_indent: usize,
```

Comment:
Can we name it to `max_indent` or `indent_limit` because this is an upper bound for indent, not a lower bound?

Reply:
Hm, no I think it is the minimum indentation for a line to be evaluated. If the minimum is 0, it means everything is included. If the minimum is usize::MAX, nothing is included.

It could also be seen as the upper bound on what is being skipped. But I find that less intuitive. I updated the comment to describe in terms of what is included instead of what is filtered.